composition-js/src/merging/merge.ts

```diff
@@ -815,6 +819,95 @@ class Merger {
     return this.metadata(sourceIdx).isFieldShareable(field);
   }
 
+  private isOverride(_sourceIdx: number, field: FieldDefinition<any>): boolean {
+    return !!this.getOverrideDirective(field);
+  }
+
+  private getOverrideDirective(field: FieldDefinition<any>): Directive<FieldDefinition<any>> | undefined {
+    let f = field;
+    while (f !== undefined) {
+      const directive = f.appliedDirectives?.find(directive => directive.definition && directive.definition.name === overrideDirectiveSpec.name);
+      if (directive) {
+        return directive;
+      }
+      f = f.parent;
+    }
+    return undefined;
+  }
+
+  /**
+   * Validates whether or not the use of the @override directive is correct.
+   * return a list of subgraphs to ignore for the current field
+   */
+  private validateOverride(sources: (FieldDefinition<any> | undefined)[], { coordinate }: FieldDefinition<any>): string[] {
+    // For any field, we can't have more than one @override directive
+    type MappedValue = {
+      idx: number,
+      name: string,
+      overrideDirective: Directive<FieldDefinition<any>> | undefined,
+    };
+
+    type ReduceResultType = {
+      subgraphsWithMoving: string[],
+      subgraphMap: { [key: string]: MappedValue },
+    };
+
+    const subgraphsToIgnore: string[] = [];
+    // convert sources to a map so we don't have to keep scanning through the array to find a source
+    const { subgraphsWithMoving, subgraphMap } = sources.map((source, idx) => {
+      if (!source) {
+        return undefined;
+      }
+      return {
+        idx,
+        name: this.names[idx],
+        overrideDirective: this.getOverrideDirective(source),
+      };
+    }).reduce((acc: ReduceResultType, elem) => {
+      if (elem !== undefined) {
+        acc.subgraphMap[elem.name] = elem;
+        if (elem.overrideDirective !== undefined) {
+          acc.subgraphsWithMoving.push(elem.name);
+        }
+      }
+      return acc;
+    }, { subgraphsWithMoving: [], subgraphMap: {} });
+
+    subgraphsWithMoving.forEach((subgraphName) => {
+      const { overrideDirective } = subgraphMap[subgraphName];
+      const sourceSubgraphName = overrideDirective?.arguments()?.from;
+      if (!this.names.includes(sourceSubgraphName)) {
+        this.hints.push(new CompositionHint(
+          hintFromSubgraphDoesNotExist,
+          `Source subgraph '${sourceSubgraphName}' for field '${coordinate}' on subgraph '${subgraphName}' does not exist`,
```

Comment:
On that particular hint, it could be nice to use the code from `internals-js/src/suggestions.ts` to check if the user might not have likely typoed the subgraph name and offer fix suggestions. Because while there may be rare cases where you use `@override` to sunset a full subgraph and thus ends up with some left-over `@override` point to a subgraph that don't exist anymore, I think the vast majority of case where user are going to get that message will be typos in the subgraph name.